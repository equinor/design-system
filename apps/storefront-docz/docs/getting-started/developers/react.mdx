---
title: Developers
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
