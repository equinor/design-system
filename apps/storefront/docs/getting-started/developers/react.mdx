---
title: Developers
metaTitle: Developers react
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
