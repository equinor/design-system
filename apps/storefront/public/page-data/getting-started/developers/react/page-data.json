{"componentChunkName":"component---docs-getting-started-developers-react-mdx","path":"/getting-started/developers/react/","result":{"pageContext":{"frontmatter":{"title":"Developers","metaTitle":"Developers react","tabs":["Introduction","CSS framework","React","Figma"],"mode":"draft","route":"/getting-started/developers/react/"},"entry":{"id":"6f9784da00517d57e4627269567073e6","filepath":"docs/getting-started/developers/react.mdx","fullpath":"/Users/vnys/Kode/storefront-replace/apps/storefront/docs/getting-started/developers/react.mdx","route":"/getting-started/developers/react/","slug":"docs-getting-started-developers-react","name":"React","menu":"","headings":[]}}}}