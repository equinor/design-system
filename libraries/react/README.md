@eds-ui/react
===============

1. `npm i -g pnpm`
1. `pnpm install`
1. `pnpm recursive install`

## Run storybook

`pnpm start`

## Build modules

`pnpm recursive run build`
