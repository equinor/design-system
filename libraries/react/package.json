{
  "name": "@eds-ui/react",
  "private": true,
  "version": "0.0.1",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "jest",
    "storybook": "start-storybook"
  },
  "keywords": [],
  "author": "",
  "license": "AGPL",
  "dependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6"
  },
  "devDependencies": {
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.4",
    "@babel/preset-env": "^7.4.4",
    "@babel/preset-react": "^7.0.0",
    "@storybook/react": "^5.0.11",
    "babel-plugin-styled-components": "^1.10.0",
    "enzyme": "^3.9.0",
    "enzyme-adapter-react-16": "^1.12.1",
    "jest": "^24.8.0",
    "jest-styled-components": "^6.3.1",
    "rollup": "^1.11.3",
    "rollup-plugin-babel": "^4.3.2",
    "rollup-plugin-commonjs": "^9.3.4",
    "rollup-plugin-node-resolve": "^4.2.3",
    "styled-components": "^4.2.0"
  },
  "peerDependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "styled-components": "^4.2.0",
    "prop-types": "^15.7.2"
  },
  "jest": {
    "verbose": true,
    "setupFilesAfterEnv": [
      "<rootDir>/testSetup.js"
    ],
    "testPathIgnorePatterns": [
      "<rootDir>/(node_modules)"
    ],
    "transformIgnorePatterns": [
      "<rootDir>/(node_modules)"
    ],
    "haste": {
      "providesModuleNodeModules": [
        ".*"
      ]
    },
    "roots": [
      "packages/node_modules"
    ]
  }
}
