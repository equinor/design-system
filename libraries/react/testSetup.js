import React from 'react'
import Enzyme from 'enzyme'
import 'jest-styled-components'
import styled from 'styled-components'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })
