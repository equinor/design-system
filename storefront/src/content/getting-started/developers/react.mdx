---
title: React
tabs:
    - Introduction
    - CSS framework
    - React
    - Figma
    - Contributing
mode: draft
---


:warning: Information coming soon
