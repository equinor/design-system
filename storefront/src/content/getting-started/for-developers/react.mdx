---
title: React
tabs:
    - Introduction
    - CSS framework
    - React
    - Figma
    - Contributing
---

Developers React
