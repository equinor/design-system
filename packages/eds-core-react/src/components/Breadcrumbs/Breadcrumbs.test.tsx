/* eslint-disable no-undef */
import { render, fireEvent, act, screen } from '@testing-library/react'
import { axe } from 'jest-axe'
import styled from 'styled-components'
import { Breadcrumbs } from '.'

const { Breadcrumb } = Breadcrumbs

const StyledBreadcrumbs = styled(Breadcrumbs)`
  position: absolute;
`

describe('Breadcrumbs', () => {
  it('Matches snapshot', () => {
    const { asFragment } = render(
      <Breadcrumbs>
        <Breadcrumb>Label 1</Breadcrumb>
        <Breadcrumb>Label 2</Breadcrumb>
        <Breadcrumb>Label 3</Breadcrumb>
      </Breadcrumbs>,
    )
    expect(asFragment()).toMatchSnapshot()
  })
  it('Should pass a11y test', async () => {
    const { container } = render(
      <Breadcrumbs>
        <Breadcrumb href="#">Label 1</Breadcrumb>
        <Breadcrumb href="#">Label 2</Breadcrumb>
        <Breadcrumb href="#">Label 3</Breadcrumb>
      </Breadcrumbs>,
    )
    expect(await axe(container)).toHaveNoViolations()
  })
  it('has aria label', () => {
    render(
      <Breadcrumbs>
        <Breadcrumb>Label 1</Breadcrumb>
        <Breadcrumb>Label 2</Breadcrumb>
        <Breadcrumb>Label 3</Breadcrumb>
      </Breadcrumbs>,
    )
    const breadcrumbs = screen.getByLabelText('breadcrumbs')

    expect(breadcrumbs).toHaveAttribute('aria-label', 'breadcrumbs')
  })
  it('should render three normal breadcrumbs', () => {
    render(
      <Breadcrumbs>
        <Breadcrumb>Label 1</Breadcrumb>
        <Breadcrumb>Label 2</Breadcrumb>
        <Breadcrumb>Label 3</Breadcrumb>
      </Breadcrumbs>,
    )

    expect(screen.getAllByRole('listitem', { hidden: false })).toHaveLength(3)
    expect(screen.getByRole('list')).toHaveTextContent(
      'Label 1/Label 2/Label 3',
    )
  })
  it('should render ellipsis between first and last crumb when collapsed', () => {
    render(
      <Breadcrumbs collapse>
        <Breadcrumb>Label 1</Breadcrumb>
        <Breadcrumb>Label 2</Breadcrumb>
        <Breadcrumb>Label 3</Breadcrumb>
        <Breadcrumb>Label 4</Breadcrumb>
        <Breadcrumb>Label 5</Breadcrumb>
        <Breadcrumb>Label 6</Breadcrumb>
      </Breadcrumbs>,
    )

    expect(screen.getAllByRole('listitem', { hidden: false })).toHaveLength(3)
    expect(screen.getByRole('list')).toHaveTextContent('Label 1/…/Label 6')
  })
  it('should expand from collapsed state when clicking the ellipsis', () => {
    render(
      <Breadcrumbs collapse>
        <Breadcrumb>Label 1</Breadcrumb>
        <Breadcrumb>Label 2</Breadcrumb>
        <Breadcrumb>Label 3</Breadcrumb>
        <Breadcrumb>Label 4</Breadcrumb>
        <Breadcrumb>Label 5</Breadcrumb>
        <Breadcrumb>Label 6</Breadcrumb>
      </Breadcrumbs>,
    )
    const ellipsisButton = screen.getByRole('button')
    expect(screen.getByRole('list')).toHaveTextContent('Label 1/…/Label 6')
    fireEvent.click(ellipsisButton)
    expect(screen.getByRole('list')).toHaveTextContent(
      'Label 1/Label 2/Label 3/Label 4/Label 5/Label 6',
    )
  })
  it('should crop the labels and have tooltip on hover when maxWidth is defined', async () => {
    const mockResizeObserver = jest.fn(() => ({
      observe: jest.fn(),
      disconnect: jest.fn(),
      unobserve: jest.fn(),
    }))
    window.ResizeObserver = mockResizeObserver
    render(
      <Breadcrumbs>
        <Breadcrumb maxWidth={30}>Label 1</Breadcrumb>
        <Breadcrumb maxWidth={30}>Label 2</Breadcrumb>
        <Breadcrumb maxWidth={30}>Label 3</Breadcrumb>
      </Breadcrumbs>,
    )
    const crumb = screen.getAllByText(/^Label\s\d$/)
    expect(crumb[0]).toHaveStyleRule('max-width', '30px')

    fireEvent.mouseEnter(crumb[0])

    await act(() => new Promise((r) => setTimeout(r, 100)))

    const tooltip = screen.getByRole('tooltip')
    expect(tooltip).toBeDefined()
  })
  it('can extend the css for the component', () => {
    render(
      <StyledBreadcrumbs>
        <Breadcrumb>Label</Breadcrumb>
      </StyledBreadcrumbs>,
    )

    expect(screen.getByLabelText('breadcrumbs')).toHaveStyleRule(
      'position',
      'absolute',
    )
  })
})
