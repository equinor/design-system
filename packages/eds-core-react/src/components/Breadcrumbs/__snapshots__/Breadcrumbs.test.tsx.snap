// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs Matches snapshot 1`] = `
<DocumentFragment>
  .c2 {
  margin: 0;
  color: var(--eds_paragraph__body_short_color, rgba(61, 61, 61, 1));
  font-family: Equinor;
  font-size: 1.000rem;
  font-weight: 400;
  line-height: 1.250em;
  text-align: left;
}

.c0 {
  list-style: none;
  display: flex;
  gap: 0.45em 0;
  align-items: center;
  padding: 0;
  margin: 0;
  flex-wrap: wrap;
}

.c1 {
  display: block;
  min-width: 30px;
}

.c4 {
  color: var(--eds_text__static_icons__tertiary, rgba(111, 111, 111, 1));
  padding-left: 16px;
  padding-right: 16px;
  display: block;
  line-height: 1;
  display: flex;
}

.c4>svg {
  margin-inline: -9px;
}

.c3 {
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  display: block;
  line-height: 1;
  text-decoration: none;
  color: var(--eds_text__static_icons__tertiary, rgba(111, 111, 111, 1));
  width: 100%;
}

@media (hover:hover) and (pointer:fine) {
  .c3:hover {
    text-decoration: underline;
    color: var(--eds_interactive_primary__resting, rgba(0, 112, 121, 1));
    cursor: pointer;
  }
}

<nav
    aria-label="breadcrumbs"
  >
    <ol
      class="c0"
    >
      <li
        class="c1"
      >
        <span
          class="c2 c3"
        >
          Label 1
        </span>
      </li>
      <li
        aria-hidden="true"
      >
        <p
          class="c2 c4"
        >
          /
        </p>
      </li>
      <li
        class="c1"
      >
        <span
          class="c2 c3"
        >
          Label 2
        </span>
      </li>
      <li
        aria-hidden="true"
      >
        <p
          class="c2 c4"
        >
          /
        </p>
      </li>
      <li
        class="c1"
      >
        <span
          class="c2 c3"
        >
          Label 3
        </span>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;
